docs/book/provider_implementations/building_running_and_testing.md

```diff
@@ -0,0 +1,60 @@
+# Building
```

Comment:
I believe this needs to go under docs/book/src/... and you need to make sure docs/book/src/SUMMARY.md is updated to include the new content. Right @vincepri?